### PR TITLE
Add Pipeline Tutorial

### PR DESCRIPTION
**Pull request by willis102**
Adding the pipeline tutorial from https://support.shotgunsoftware.com/hc/en-us/articles/219039938-Pipeline-Tutorial.

### CODE REVIEW

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/6bd9fdf7be48300284d52eb96e9ab02f79e13a15/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/7bf4441d1bf7dc7a92df2ebc4f4f0506d7a99973/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/7600d478d725d1d58b02593914accb4e905bd295/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/551869f5bdfff271948184f90c682b620355da19/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/4ca163d2a6bb818542016564f11e5f9679c81e49/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/f0e74712cc2374ac83978d4200ca04435cee6cd2/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/52ef622149b5bf965b754549c776bf1e80bff558/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/72590e4834f025810c5436ae08c337dc77d1a814/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/79e9a9513ec25e4d40615b3b8281c2b3fc6f33c7/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/ed3c8ceba03db92400ac4432d0a9753c582f940d/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/1fade7ef4ac532029beddb625cb88defd2d1baed/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/063e0c58b538b59e038dbf547ca2512f4eab9e30/index.html)

**Comment by willis102** [comment]
I believe the congent is correctly copied over and formatted, however I think we may want to add some padding-bottom to `img` (or maybe just `figure > image`) in the just-the-docs SASS.  Let me know if you agree, and I can do that before merging.

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/5ea2599b1d7971b6a46dbcf0f3d1895a1a6a8f91/index.html)

File: docs/en/toolkit/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,835 @@
...
+
+Once Maya has full loaded, you will see the File Open Dialog appear. This dialog allows you to browse existing Maya files within the project. It also allows you to create new files that the Shotgun integrations will be aware of. 
+
+Select the Assets tab and drill down into the Teapot's modeling task. Since there are no artist work files for this task yet, click the **+ New File** button. 
+
+{% include figure src="../../images/tk_pipeline_tutorial/image_13.png" %}
+
+Clicking this button will create a new, empty Maya session and set your current working context to the Teapot Asset's Model task. 
+
+NOTE: At any time during this tutorial you can launch the Shotgun Panel via the Shotgun menu in Maya or Nuke. This panel provides a view into your project data without leaving your DCC. It will show you your current working context and any recent activity within that context. You can also add notes for feedback directly into the panel. See the [Shotgun Panel documentation](https://support.shotgunsoftware.com/hc/en-us/articles/115000068574-Integrations-user-guide#The%20Shotgun%20Panel) for more info.
```
Comment (manneohrstrom) [inline docs/en/toolkit/tutorial.md]: can this be a note box?

File: docs/en/toolkit/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,835 @@
...
+
+This tutorial covers building a simplified, yet typical, pipeline for animation or visual effects production. By following this tutorial you will build a pipeline that provides all of the pieces necessary to push Assets from modeling through look development, and then into and through a production scene. 
+
+Much of the workflows covered in this pipeline work out-of-the-box with Shotgun's built-in integrations. For the portions of the pipeline where studios are more often building custom solutions the tutorial will walk you through the process of customizing the artists workflow using the Toolkit platform.
+
+Here is a high level view of the pipeline you will build in this tutorial: 
+
+{% include figure src="../../images/tk_pipeline_tutorial/image_0.png" caption="Pipeline Overview" %}
+
+Pipeline Overview
```
Comment (manneohrstrom) [inline docs/en/toolkit/tutorial.md]: this looks weird - add new heading?

File: docs/en/toolkit/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,835 @@
...
+
+### Templates
+
+This tutorial also uses the templates defined in the Default pipeline configuration. You can open the **`config/core/templates.yml`** file to get a peek at the templates that are used by the apps to map input and output files to paths on disk. For more information about the templating system, see the [File System Configuration Reference](https://support.shotgunsoftware.com/hc/en-us/articles/219039868) documentation.
+
+### Hooks
+
+Much of this tutorial will involve modifying App hooks in order to customize the artist workflows. Before diving into that customization, you should have a basic understanding of what hooks are, how they work, and where they live. Read through the Hooks section of the  [Administration and ](https://support.shotgunsoftware.com/hc/en-us/articles/219033178#Hooks)[Configuration](https://support.shotgunsoftware.com/hc/en-us/articles/219033178#Hooks) documentation.
```
Comment (manneohrstrom) [inline docs/en/toolkit/tutorial.md]: the two links look weird - move `and` out of the hyperlink?

File: docs/en/toolkit/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,835 @@
...
+
+# Building the Pipeline
+
+At this point you should be ready to begin building a pipeline. You have a project set up in Shotgun, you can launch Maya & Nuke via Desktop, and you've taken control of the project's configuration. You also have a basic understanding of the structure of the config and are ready to begin fleshing out the artist workflow. 
+
+The following sections will walk through each step of the pipeline, highlighting the features that are available out-of-the-box and walking you through the process of customizing the Shotgun integrations. By the end of these sections, you will have a simple, fully functional, end-to-end production pipeline. You will also get a feel for the steps artists will take as they work on production.
+
+**NOTE: All code and configuration for this tutorial can be found on the **`pipeline_tutorial`** branch of the [**`tk-config-default2`** repository](https://github.com/shotgunsoftware/tk-config-default2/tree/pipeline_tutorial/). Feel free to use this branch if you need hints as to where files should live, where code should be added, etc.**
```
Comment (manneohrstrom) [inline docs/en/toolkit/tutorial.md]: some weird formatting here!

File: docs/en/toolkit/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,835 @@
...
+
+Once you have saved the Teapot model to version 2, you are ready for the last step in this section of the tutorial. 
+
+Now that your Teapot model is ready, you need to publish it so that it can be surfaced and rigged. To publish, click the **Shotgun > Publish…** menu action. You will be presented with the Publish App dialog. 
+
+{% include figure src="../../images/tk_pipeline_tutorial/image_18.png" %}
+
+The dialog shows a tree of items representing what will be published. The tree includes some entries that represent the items to be published and some entries represent the actions that will be performed during the publish operation. 
+
+On the left side of the dialog you will see an item representing the current Maya session. Underneath it, you will see a **Publish to Shotgun **child action. An additional item representing **All Session Geometry** is shown as a child item of the current session. It also has a **Publish to Shotgun** child action.
```
Comment (manneohrstrom) [inline docs/en/toolkit/tutorial.md]: weird formatting

File: docs/en/toolkit/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,835 @@
...
+
+Shader management can be a time consuming and complex task when building a pipeline. It is often very specific to a studio. It is for these reasons that the shipped Maya integration does not handle shader or texture management out-of-the-box. 
+
+Use the **Shotgun > File Save…** menu action to save the current session before continuing.
+
+### Custom Shader Publish
+
+For the purposes of this simple pipeline, you will customize the Publisher app to export Maya shader networks as additional publish items from the surfacing step. Later in the tutorial, you will put together a quick and dirty solution that allows the shaders to be reconnected to the Alembic geometry caches when referenced downstream.
+
+**NOTE: The customization you'll be adding is, admittedly, very simple and fragile. A more robust solution might take into account alternate representations of a surfaced character as well as the asset management side of using external images as texture maps. This example presents only a starting point for building a real-world solution.**
```
Comment (manneohrstrom) [inline docs/en/toolkit/tutorial.md]: note box

**Comment by manneohrstrom** [comment]
Also, before we merge, we need to ask @tannazie to find out the right permanent home for this in the TOC - right now we just wedged it in where it seemed roughly right. I'll add her as a reviewer so she can give feedback on the appropriate location.

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/30b30a9db954cd9c2ff01ecf9344894b0f592105/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/3501e9be95b19815b36dcd49cfa023eee5fd60fe/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/c644c5ad1486a13cf9d3e22d13cc6175f84c36cf/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/730d7b4f60f8babc045460fc3d41b18b6bc04879/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/5cb723749f1ee779eaf2bbf19d1042873e761c6d/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/6b49fa56884e21022140a6e8f87d4aa29bc82e62/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/0c579024499109255c7dbf5d1fa2f9b9c4ee8baa/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/fcb6b9a8cce32e31b79a4adb2dffdea7a8b1845d/index.html)

**Comment by willis102** [comment]
> Looks great! See comments - seems there a are few places where we can use an info box? Also noticed some funny formatting.
Added info boxes everywhere it seemed appropriate.  We'll need to merge in some changes in just-the-docs to get the markdown inside includes to work correctly.

> Are you talking about adding some spacing below the pictures, like here?
> image
> That sounds great!
Added!  I think it looks better like this.  Let me know what you think.

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/33a3c459644aad036cd9026e141a59731cd4812d/index.html)

**Comment by tannazie** [comment]
This is looking so awesome!  On where to put it:

I'm thinking something like this:
Toolkit

- Learning Resources
  - Guides
     - 1
     - 2
     - ….
     - 6
 - Tutorial
 - Video
 - Whatever else

I don’t know if we can have that level of nesting, but having a section that all the guides, tutorials, “learning content” go into seems like the right move. @manneohrstrom thoughts?

And one small note, this looks a little wonky: 

Pipeline_Tutorial_-_SG_Developer

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/d061ddd05e51861e22cdb97fa74f7c83ed52729e/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/7f9f76e93d5eea265b90da8f15697adf86929e66/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/9ca4f2a85e995233877384198965f77bc9f9acff/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/bed44733de9dbc9e156f3da8610947765d8a9f98/index.html)

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/c29be75d9fc213237c0a9ac0845b1e920365f9fa/index.html)

**Comment by willis102** [comment]
- Added python syntax highlighting.
- Reorganized per @tannazie's comment
- Removed the images @tannazie pointed out that were looking wonky

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/f224d4c4ccd7556cc3c17880820a5172eb7dee2e/index.html)

**Comment by manneohrstrom** [comment]
@willis102 perfect, yes it's the right move! 😄 Looking now!

**Comment by willis102** [comment]
> * Looks like the automatically generated "child pages" chunk that normally goes at the bottom of the page is missing:

Ah, I missed that when I was adding support for deeper child pages.  Seems to be back now:
http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/d4e1c10a0082f7c17c15a074dadb23408a67d7d8/toolkit/learning-resources/guides/

> * When pipeline tutorial is selected, the child items for basic guides is expanded. That seems odd to me - i _think_ it makes more sense if only the child items of the currently selected child (or grand child) is expanded. Effectively, being consistent with how the parenting level above works. I think the expansion behaviour should be consistent!

Makes sense, should be fixed:
http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/d4e1c10a0082f7c17c15a074dadb23408a67d7d8/toolkit/learning-resources/tutorial/

**Comment by manneohrstrom** [comment]
[Documentation Preview](http://doctest3.s3-website.eu-west-2.amazonaws.com/tk-doc-generator/d90c2ffb12650f4b4a43971adb19a3e2717eff58/index.html)

**Pull request merged**